File: javascript/easy/0217.containsDuplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function (nums) {
  const numCounter = {};
  let hasDuplicate = false;

  for (let i = 0; i < nums.length; i++) {
    if (numCounter[nums[i]]) {
      numCounter[nums[i]]++;
      hasDuplicate = true;
    } else {
      numCounter[nums[i]] = 1;
    }
  }

  return hasDuplicate == true ? true : false;
};

console.log(containsDuplicate([1, 2, 3, 1])); // true
// Given an integer array nums, return true if any value appears at least twice in the array,
// and return false if every element is distinct.

console.log(containsDuplicate([1, 2, 3, 4])); // false

console.log(containsDuplicate([1, 1, 1, 3, 3, 4, 3, 2, 4, 2])); // true

File: python/easy/0217.containsDuplicate.py
from typing import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        num_counter = {}
        for number in nums:
            if number in num_counter:
                num_counter[number] += 1
                return True
            else:
                num_counter[number] = 1

        return False


if __name__ == "__main__":
    solution = Solution()
    print(solution.containsDuplicate([1, 2, 3, 1]))  # true
    print(solution.containsDuplicate([1, 2, 3, 4]))  # false
